assets/js/bulk-move.js

```diff
@@ -8,10 +8,46 @@
 
 /*jslint browser: true, devel: true*/
 /*global BULK_MOVE, jQuery, document, postboxes, pagenow, ajaxurl*/
+BULK_MOVE.validate_same_user_roles = function(that) {
+
+    var fromUserRole = jQuery( '#bm-users-by-role-from-roles-list' ).val().toLowerCase(),
+        ToUserRole = jQuery( '#bm-users-by-role-to-roles-list' ).val().toLowerCase();
+
+    fromUserRole = jQuery.trim( fromUserRole );
+    ToUserRole = jQuery.trim( ToUserRole );
+
+    return fromUserRole === ToUserRole;
+};
+
 jQuery(document).ready(function () {
-	jQuery( 'button[value="move_tags"], button[value="move_cats"], button[value="move_category_by_tag"], button[value="move_custom_taxonomy"]' ).click( function () {
-		return confirm( BULK_MOVE.msg.move_warning );
-	});
+
+    jQuery('button[value="move_tags"], button[value="move_cats"], button[value="move_category_by_tag"], button[value="move_custom_taxonomy"], button[value="move_users_by_role"]').click(function (e) {
+
+        var currentButton = jQuery(this).val(),
+            valid = true,
+            that = this,
+            errorKey;
+
+        if (!currentButton in BULK_MOVE.validators) {
+            return confirm(BULK_MOVE.msg.move_warning);
+        }
+
+        jQuery.each(BULK_MOVE.validators[currentButton], function (index, validator) {
+            valid = valid && (!BULK_MOVE[validator](that));
+
+            if (!valid) {
+                errorKey = validator.replace('validate_', '');
+                alert(BULK_MOVE['error'][errorKey]);
+                return false;
+            }
+        });
+
+        if ( valid ) {
+            return confirm(BULK_MOVE.msg.move_warning);
+        } else {
+            e.preventDefault();
```

Comment:
There is no return statement in this branch, but you do return something in other branches.

Reply:
> There is no return statement in this branch, but you do return something in other branches.

Since this is the last statement, I didn't add it explicitly.